### ISSUE TITLE
Empty Atom/RSS feed for protobuf.dev/news

### ISSUE THREAD

**Issue by ejona86** [protocolbuffers/protobuf#12446]
Recent release notes keep mentioning [Protobuf News](https://protobuf.dev/news/) as a source of important news. But how are we supposed to learn there are new news items? It seems it is just a bunch of bullets on a page.

In the HTML, Hugo is including alternate reference to an RSS feed, but [the feed has no entries](https://protobuf.dev/news/index.xml). Is that a bug? Or is there an alternate expected form of notification?

**Comment by Logofile** [comment]
Thanks for raising this issue. I'm looking into how to get Hugo/Docsy to populate the file at [protobuf.dev/news/index.xml](protobuf.dev/news/index.xml). It's turning into a bigger effort than it should be.

**Comment by ejona86** [comment]
This is still an issue.

**Comment by Logofile** [comment]
I agree. This was closed by a bot, but it's still on my to-do list to address. I did do some research into this, but need to pick it back up.

**Comment by ejona86** [comment]
Without any way to receive news notifications, News is not a method of announcement. That's a real shame because it is clear effort is being put into them.

Right now they feel more like a notice "on display in the bottom of a locked filing cabinet stuck in a disused lavatory."

**Comment by Logofile** [comment]
I swear that I've spent several hours trying to fix this! But your wonderful humor has inspired me to give it another go. Wish me luck!

**Comment by ejona86** [comment]
The [default RSS template](https://github.com/gohugoio/hugo/blob/master/tpl/tplimpl/embedded/templates/_default/rss.xml) finds the news items. But the [docsy RSS template](https://github.com/google/docsy/blob/main/layouts/_default/list.rss.xml) doesn't use `.Pages` or `.RegularPages` _when the parent of the section is Home_ and instead loops over the entire site's pages and filters by type: `where $.Site.RegularPages "Type" $.Section`. Unclear why it does that, but it seems it would generally be fine.

But that doesn't work because the news items all have `type = "docs"`. Which... seems strange. It really seems like there should be `news/chrony/` and `news/release/` sections, and add `date` to news items to sort them. Since no date is set, removing `type = "docs"` is not enough, because the pages are in a random order and thus you can't show "the 50 most recent news items."

Backfilling dates is annoying but simple. But I don't know whether y'all would prefer making a new RSS template that uses `.Pages` (least invasive approach) or removing `type = "docs"` (probably best long-term, but unclear why that was done originally).

**Comment by Logofile** [comment]
This is **exceedingly** helpful information! I was actually talking to one of the creators of the Docsy theme, and even they didn't know about (or at least didn't mention) the `type = "docs"` issue. I'll play around with removing that and see if that affects the RSS feed in our implementation.

**Comment by Logofile** [comment]
This is still on my radar. I will keep this issue open until it is resolved.

**Comment by ejona86** [comment]
This is still an issue.

**Comment by Logofile** [comment]
Well... we're **closer**. https://protobuf.dev/index.xml has an RSS feed, but the dates got reset to a default value somewhere in the processing. Let's keep this issue open until that sub-issue is resolved.

**Comment by ejona86** [comment]
Since this is still not fixed, and apparently there is some interest still in fixing it, can this be reopened?

**Comment by Logofile** [comment]
I hadn't realized it had been closed; thanks for pointing that out. I've reopened it.

**Comment by ejona86** [comment]
This needs to remain open to handle item dates. While the feed is no longer empty, without dates the feed won't be able to order by age to make sure to include new items.

**Comment by ejona86** [comment]
Still an issue.

**Comment by ejona86** [comment]
Still an issue.

**Comment by ejona86** [comment]
Still an issue. (Pre-empting the bot by a few days)

**Comment by jguamie** [comment]
@PilgrimMemoirs will be looking into this.
